packages/api/src/webhooks/cannon.ts

```diff
@@ -585,6 +605,24 @@ export default class WebhookCannon {
   }
 }
 
+async function checkUrlExists(url) {
+  try {
+    let params = {
+      method: "HEAD",
+      timeout: 5 * 1000,
+    };
+    const resp = await fetchWithTimeout(url, params);
+
+    if (resp.status == 200) {
+      return true;
+    }
+  } catch (e) {
+    console.log("error fetching " + url, e);
+    return false;
```

Comment:
_Maybe_ only return `false` if this is a timeout error, otherwise we might wrongly fallback to the secondary when the file did exist but there was a temporary error or something. 

Actually, a temporary network issue could result in a timeout as well, so maybe we don't want to `catch` anything here after all and just let exceptions be themselves (lol sorry for the back and forth, I know I just suggested it). On an exception, [this code](https://github.com/livepeer/livepeer-com/blob/b4b41a178b814008fe79b2298ac506d4115fae63/packages/api/src/webhooks/cannon.ts#L103) will actually `nack` the AMQP message to be retried later.

Reply:
ah of course yep makes sense